includes/Checker/Checks/Plugin_Readme_Check.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks\Plugin_Readme_Check
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker\Checks;
+
+use WordPress\Plugin_Check\Checker\Check_Result;
+
+/**
+ * Check the plugins readme.txt file and contents.
+ *
+ * @since n.e.x.t
+ */
+class Plugin_Readme_Check extends Abstract_File_Check {
+
+	/**
+	 * Check the readme.txt file.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @param array        $files  Array of plugin files.
+	 */
+	protected function check_files( Check_Result $result, array $files ) {
+		// Find the readme file.
+		$readme = self::filter_files_by_regex( $files, '/readme\.txt$/' );
+
+		// If the readme.txt does not exist, add a warning and skip other tests.
+		if ( empty( $readme ) ) {
+			$result->add_message(
+				false,
+				__( 'The plugin readme.txt does not exist.', 'plugin-check' ),
+				array(
+					'file' => 'readme.txt',
+					'code' => 'no_plugin_readme',
+				)
+			);
+
+			return;
+		}
+
+		// Check the readme.txt for default text.
+		$this->check_default_text( $result, $readme );
+
+		// Check the readme.txt for a valid license.
+		$this->check_license( $result, $readme );
+
+		// Check the readme.txt for a valid version.
+		$this->check_stable_tag( $result, $readme );
+	}
+
+	/**
+	 * Checks the readme.txt for default text.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @param array        $files  Array of plugin files.
+	 */
+	private function check_default_text( Check_Result $result, array $files ) {
+		if (
+			self::file_str_contains( $files, 'Here is a short description of the plugin.' ) ||
+			self::file_str_contains( $files, 'Tags: tag1' ) ||
+			self::file_str_contains( $files, 'Donate link: http://example.com/' )
+		) {
+			$result->add_message(
+				false,
+				__( 'The readme.txt appears to contain default text.', 'plugin-check' ),
+				array(
+					'code' => 'default_readme_text',
+					'file' => $result->plugin()->path( '/readme.txt' ),
+				)
+			);
+		}
+	}
+
+	/**
+	 * Checks the readme.txt for a valid license.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The Check Result to amend.
+	 * @param array        $files  Array of plugin files.
+	 */
+	private function check_license( Check_Result $result, array $files ) {
+		if ( ! self::file_preg_match( '/(License:\s*([a-z0-9\-\+\.]+)(\sor\s([a-z0-9\-\+\.]+))*)|(License URI:\s*([a-z0-9\-\+\.]+)(\sor\s([a-z0-9\-\+\.]+))*)/i', $files ) ) {
```

Comment:
A lot of this regex is unnecessarily duplicated I think. The value part (i.e. the part of the regex that you copied over from the other tool) could be only included once, and then you only use an "OR" for the field name (something like `(License:|License URI:)`).

Reply:
It was addressed in https://github.com/10up/plugin-check/pull/191/commits/e832dfce38d83d396351736be2b0faec5e2eefd8